beta/nncf/tensorflow/layers/wrapper.py

```diff
@@ -65,14 +66,77 @@ def trainable(self):
     @trainable.setter
     def trainable(self, value):
         self.layer.trainable = value
+        # Trainable property of all operations follows the global trainable property of NNCFWrapper
+        self.set_ops_trainable(value)
+
+    def set_ops_trainable(self, value):
+        for ops in self.weights_attr_ops.values():
+            for op in ops.values():
+                op.trainable = value
 
     @property
     def trainable_weights(self):
-        return self._trainable_weights + self.layer.trainable_weights
+        trainable_weights_by_definition = self._trainable_weights + self.layer.trainable_weights
+
+        if self._op_build:
+            trainable_ops_weights_on, trainable_ops_weights_off = self.classify_trainable_ops_weights()
+        else:
+            trainable_ops_weights_on, trainable_ops_weights_off = [], []
+
+        if self.trainable:
+            weights = trainable_weights_by_definition
+            for trainable_weight_off in trainable_ops_weights_off:
+                weights.remove(trainable_weight_off)
+            outputs = weights
+        else:
+            outputs = trainable_ops_weights_on
```

Comment:
I'm a bit confused that layer which has attribute `trainable` = `False` returns some weights by default. So, if I want
1) Only freeze wrapped layer weight I have to set `layer.trainable = False`
2) Only operation weights I have to set `op.trainable = False` for each operation I want to freeze
3) To freeze whole layer I have to both set `layer.trainable = False` and  `op.trainable = False` for each operation I want to freeze. 

If I would decided to freeze whole layer I would expect that I only need to specify  `layer.trainable = False. I think it would be better to freeze wrapped layer weight and operation weight by separate attributes (like `trainable_wrapped_layer` or something like that) and use `layer.trainable` as which freeze/unfreeze all layer weight. We should discuss this.

Reply:
Let me clarify.
For now, the default behavior is as follows. If you set `layer.trainable` some value, `op.trainable` will be set the same value. 

Therefore:
1. _Freeze_ both wrapped layer weights and op weights -> `layer.trainable = False`
2. _Unfreeze_ both wrapped layer weights and op weights -> `layer.trainable = True`

Introducing `trainable `property to the operations, we have the following additional options:
3. _Frozen_ wrapped layer weights but _unfrozen_ op weights -> `layer.trainable = False; op.trainbale = True`
4. _Unfrozen_ wrapped layer weights but _frozen_ op weights -> `layer.trainable = True; op.trainbale = False`

Operations have the same `trainable` property as the whole layer because this property follows a similar logic.

Reply:
Yes, I missed that `ops.trainable` have the same value as `layer.trainable`. That's actually make sense, thank you for explanation! By the way, I think keeping this instruction somewhere in docstring would be useful for `NNCFWrapper` users

Reply:
But it still tricky for me because `trainable` attribute isn't just the state of the layer weights but contain its own state inside. For example 
```
layer.trainable = False
...
ops.trainable = True
``` 
leads to layer weight freezing and operation weights training. But
```
ops.trainable = False
...
layer.trainable = True
```
leads to complete freezing. You can see that the logic of trainable attribute become more complex than logic of indicator variable

Reply:
Sure, I will create necessary docstrings